src/main/java/step5/domain/RacingGame.java

```diff
@@ -0,0 +1,43 @@
+package step5.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import step5.util.CarFactory;
+
+public class RacingGame {
+
+    private final List<Car> cars = new ArrayList<>();
+    private final List<RacingResult> racingResults = new ArrayList<>();
+    private final int tryCount;
+    private final MovingStrategy movingStrategy;
+
+    public RacingGame(String[] carNames, int tryCount, MovingStrategy movingStrategy){
+        this.cars.addAll(CarFactory.createCar(carNames));
+        this.tryCount = tryCount;
+        this.movingStrategy = movingStrategy;
+    }
```

Comment:
이러한 부분도 저는 보통 필드의 타입 그대로 생성자 파라미터로 전달받고 내부에서는 직접적인 의존을 피하려고 합니다. 지금의 코드는 추후 CarFactory의 메서드 변경이 생기면 해당 도메인 코드의 변경이 같이 발생해요. 직접적인 의존을 가지고 있기 때문인데 이를 파라미터로 전달해서 의존의 정도를 낮춰준다면 변경에 있어서 좀 더 유연한 코드가 될 수 있을 것 같습니다!!

```suggestion
// RacingGame.java
public class RacingGame {
    // ... tryCount의 경우에도 꼭 필드로 관리할 필요가 없다고 느껴져서 예시에서는 제외했습니다.
    
    public RacingGame(List<Car> cars) {
        this.cars = cars;
    }
    
    // ...
}

// RacingGameController.java
public class RacingGameController {
    // ...
    
    public void play() {
        // ...
        List<Car> cars = CarFactory.createCar(carNames);
        RacingGame racingGame = new RacingGame(cars);
        List<RacingResult> results = racingGame.start(tryCount, movingStrategy);
        // ...
    }
}
```